### PR TITLE
Remove `check-query-cache` experimental flag

### LINKED ISSUE #1007: Remove `check-query-cache` experimental flag
**Summary**

Remove experimental flag and guarding around it.

### PR DESCRIPTION
**Pull request by Siddhant-K-code**
## References

fixes #1007

## Review Checklist
- [X] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: .config-schema.json [hunk 1]
```diff
@@ -78,7 +78,7 @@
             "type": "array",
             "items": {
                 "type": "string",
-                "enum": ["check-query-cache"]
+                "enum": [""]
```
Comment (Siddhant-K-code) [inline .config-schema.json:81]: Not sure, if I just remove this or remove whole `"experimentals"` from the config schema

Comment (miparnisari) [reply .config-schema.json:81]: Just remove from the enum 👍

File: cmd/openfga/main_test.go [hunk 2]
```diff
@@ -204,210 +204,6 @@ func newOpenFGATester(t *testing.T, args ...string) OpenFGATester {
 	}
 }
 
-func TestCheckWithQueryCacheEnabled(t *testing.T) {
```
Comment (miparnisari) [inline cmd/openfga/main_test.go:207]: Please keep this test, it's still applicable. The only change needed should be

```go
tester := newOpenFGATester(t, "--check-query-cache-enabled=true")
```

File: cmd/run/flags.go [hunk 3]
```diff
@@ -164,15 +164,6 @@ func bindRunFlagsFunc(flags *pflag.FlagSet) func(*cobra.Command, []string) {
...
 
-		util.MustBindPFlag("checkQueryCache.enabled", flags.Lookup("check-query-cache-enabled"))
-		util.MustBindEnv("checkQueryCache.enabled", "OPENFGA_CHECK_QUERY_CACHE_ENABLED")
-
-		util.MustBindPFlag("checkQueryCache.limit", flags.Lookup("check-query-cache-limit"))
-		util.MustBindEnv("checkQueryCache.limit", "OPENFGA_CHECK_QUERY_CACHE_LIMIT")
-
-		util.MustBindPFlag("checkQueryCache.ttl", flags.Lookup("check-query-cache-ttl"))
-		util.MustBindEnv("checkQueryCache.ttl", "OPENFGA_CHECK_QUERY_CACHE_TTL")
-
```
Comment (miparnisari) [inline cmd/run/flags.go:175]: Please keep these flags, they are still applicable

**Review by miparnisari — CHANGES_REQUESTED** [review]

File: pkg/server/server.go [hunk 4]
```diff
@@ -279,19 +276,6 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
...
-			zap.Uint32("CheckQueryCacheLimit", s.checkQueryCacheLimit))
-		s.checkCache = ccache.New(
-			ccache.Configure[*graph.CachedResolveCheckResponse]().MaxSize(int64(s.checkQueryCacheLimit)),
-		)
-		s.checkOptions = append(s.checkOptions, graph.WithCachedResolver(
-			graph.WithExistingCache(s.checkCache),
-			graph.WithCacheTTL(s.checkQueryCacheTTL),
-		))
-	}
-
```
[the remark this answers is not in the document]
Comment (Siddhant-K-code) [reply pkg/server/server.go]: Resolved in [`3b3e86d`](https://github.com/openfga/openfga/pull/1071/commits/3b3e86d7887d53bb2db128c4d2bf2733563d108d)

**Review by miparnisari — CHANGES_REQUESTED** [review]
Remove this line too please https://github.com/openfga/openfga/blob/3b3e86d7887d53bb2db128c4d2bf2733563d108d/pkg/server/server.go#L48

**Comment by Siddhant-K-code** [comment]
> Remove this line too please
> 
> https://github.com/openfga/openfga/blob/3b3e86d7887d53bb2db128c4d2bf2733563d108d/pkg/server/server.go#L48

I've removed this in 82e370f6fd7d0a7ad56a337f4208dd4bb78f2e6b

I've left things like this (to possible reuse in future experimental feature flags): 

https://github.com/Siddhant-K-code/openfga/blob/a34f909153d67f88725b587c01baec9f5f96cffa/pkg/server/server.go#L93

https://github.com/Siddhant-K-code/openfga/blob/a34f909153d67f88725b587c01baec9f5f96cffa/pkg/server/server.go#L198

https://github.com/Siddhant-K-code/openfga/blob/a34f909153d67f88725b587c01baec9f5f96cffa/pkg/server/server.go#L261

**Review by miparnisari — APPROVED** [review]

**Pull request merged**